src/Staking.sol

```diff
@@ -120,23 +159,34 @@ contract StakeRegistry is AccessControl, Pausable {
     }
 
     /**
-     * @dev Withdraw stake only when the staking contract is paused,
-     * @param _amount The amount of ERC20 tokens to be withdrawn
+     * @dev Withdraw node stake surplus
      */
-    function withdrawFromStake(uint256 _amount) external whenPaused {
+    function withdrawFromStake() external {
         Stake memory stake = stakes[msg.sender];
 
-        // We cap the limit to not be over what is possible
-        uint256 withDrawLimit = (_amount > stake.stakeAmount) ? stake.stakeAmount : _amount;
-        stake.stakeAmount -= withDrawLimit;
+        uint256 _surplusStake = stake.potentialStake -
+            calculateEffectiveStake(stake.commitedStake, stake.potentialStake);
```

Comment:
I'd rather create a separate public function for this as SWIP suggests so that clients could be sure whether they can withdraw by one call (instead of calling protentialBalance and effectiveBalance separately).

Reply:
oh, yes, makes sense that they can check first how much can they WD.